node/src/components/chain_synchronizer/operations.rs

```diff
@@ -948,34 +954,149 @@ where
     }
 }
 
+#[derive(Debug)]
+pub(crate) enum FetchAndStoreBlockError {
+    Fetcher(FetcherError<BlockAndDeploys>),
+    WrongTypeUnderDeployApprovalsRootHash(Box<StoredValue>),
+    CLValue(CLValueError),
+    Engine(engine_state::Error),
+    BytesRepr(bytesrepr::Error),
+    TrustedStateRootMissing,
+    Impossible(String),
+    NoApprovalsForDeploy(DeployHash),
+}
+
+/// Retrieves the deploy approvals root hash from global state
+async fn fetch_deploy_approvals_root_hash<REv>(
```

Comment:
I have not added a metric before -- is there an example you can point me to?

Reply:
Certainly.
You start in the `node/src/components/chain_synchronizer/metrics.rs`, when you need to add a new metric and register it in `new()`. Please decide on a name consistent with other metrics.

Then, because the `fetch_deploy_approvals_root_hash()` is a recurring operation we need to sum up time spend on all invocations; we can do it via a call to `ctx.metrics.observe_...()`.

Please have a look at `fetch_and_store_deploys()` as an example - it does exactly what we need.

Reply:
Yeah, I was fooled by the `fetch_` part of the name. If it's designed to return fast, we should not bloat the metrics.